sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -105,35 +105,38 @@ object JDBCRDD extends Logging {
    * Turns a single Filter into a String representing a SQL expression.
    * Returns None for an unhandled filter.
    */
-  def compileFilter(f: Filter): Option[String] = {
+  def compileFilter(f: Filter, dialect: JdbcDialect): Option[String] = {
     Option(f match {
-      case EqualTo(attr, value) => s"$attr = ${compileValue(value)}"
+      case EqualTo(attr, value) => s"${dialect.quoteIdentifier(attr)} = ${compileValue(value)}"
```

Comment:
Add a nested function in compileFilter 
```Scala
def quote(colName: String): String = dialect.quoteIdentifier(colName)
```

Then, your code changes can look cleaner.